clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/metrics/TimedCallable.java

```diff
@@ -0,0 +1,77 @@
+package com.netflix.spinnaker.clouddriver.metrics;
+
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.Registry;
+import groovy.lang.Closure;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+public class TimedCallable<T> implements Callable<T> {
+
+  public static TimedCallable<Void> forRunnable(Registry registry, Id metricId, Runnable runnable) {
+    return new TimedCallable<Void>(registry, metricId, new RunnableWrapper(runnable));
+  }
+
+  public static <T> TimedCallable<T> forCallable(
+      Registry registry, Id metricId, Callable<T> callable) {
+    return new TimedCallable<T>(registry, metricId, callable);
+  }
+
+  public static <T> TimedCallable<T> forClosure(
+      Registry registry, Id metricId, Closure<T> closure) {
```

Comment:
Can we get rid of this since Closure extends Callable? (leave the method but Closure<T> to Callable<T> and it gets rid of the groovy.lang import/dependency)

Reply:
Sure, I was keeping it around assuming it was something incompat with Callable.